DuckDuckGo/PhishingDetection/PhishingDetection.swift

```diff
@@ -0,0 +1,151 @@
+            revision: revision,
+            filterSetURL: filterSetURL,
+            filterSetDataSHA: filterSetDataSHA,
+            hashPrefixURL: hashPrefixURL,
+            hashPrefixDataSHA: hashPrefixDataSHA
+        )
+        self.dataStore = dataStore ?? PhishingDetectionDataStore(dataProvider: resolvedDataProvider)
+        self.detector = detector ?? PhishingDetector(apiClient: detectionClient, dataStore: self.dataStore)
+//        self.detector = detector ?? PhishingDetector(apiClient: detectionClient, dataStore: self.dataStore, eventMapping:
+//                                                        EventMapping<PhishingDetectionEvents> {event, _, params, _ in
+//            switch event {
+//            case .errorPageShown(clientSideHit: let clientSideHit):
+//                PixelKit.fire(PhishingDetectionEvents.errorPageShown(clientSideHit: clientSideHit))
+//            case .iframeLoaded:
+//                PixelKit.fire(PhishingDetectionEvents.iframeLoaded)
+//            case .visitSite:
+//                PixelKit.fire(PhishingDetectionEvents.visitSite)
+//            case .updateTaskFailed48h(error: let error):
+//                PixelKit.fire(PhishingDetectionEvents.updateTaskFailed48h(error: error))
+//            }
+//        })
+        self.updateManager = updateManager ?? PhishingDetectionUpdateManager(client: detectionClient, dataStore: self.dataStore)
+        self.dataActivities = dataActivities ?? PhishingDetectionDataActivities(phishingDetectionDataProvider: resolvedDataProvider, updateManager: self.updateManager)
+        self.detectionPreferences = detectionPreferences
+        self.featureFlagger = NSApp.delegateTyped.featureFlagger
+        self.config = AppPrivacyFeatures.shared.contentBlocking.privacyConfigurationManager.privacyConfig
+        if let featureFlagger = featureFlagger,
+           featureFlagger.isFeatureOn(.phishingDetection),
+           self.detectionPreferences.isEnabled {
+            startUpdateTasks()
+        }
+        self.setupBindings()
+    }
+
+    convenience init(
+        dataActivities: PhishingDetectionDataActivityHandling,
+        dataStore: PhishingDetectionDataSaving,
+        detector: PhishingDetecting
+    ) {
+        self.init(
+            dataStore: dataStore,
+            detector: detector,
+            dataActivities: dataActivities
+        )
+    }
+
+    private func setupBindings() {
+        cancellable = detectionPreferences.$isEnabled.sink { [weak self] isEnabled in
+            self?.handleIsEnabledChange(enabled: isEnabled)
+        }
+    }
+
+    public func checkIsMaliciousIfEnabled(url: URL) async -> Bool {
+        if config.isFeature(.phishingDetection, enabledForDomain: url.host),
+           detectionPreferences.isEnabled {
+            return await detector.isMalicious(url: url)
+        } else {
+            return false
+        }
+    }
+
+    private func handleIsEnabledChange(enabled: Bool) {
+        if enabled {
```

Comment:
I think we already make sure in BSK that start and stop do not conflict right?

Reply:
Yeah correct, we already have this in BSK:
```
    public func start() {
        if !running {
            Task {
                for scheduler in schedulers {
                    await scheduler.start()
                }
            }
            running = true
        }
    }
    ```

Technically `stop()` can be called twice, but that should just do nothing since the task will be nil. i.e. it's idempotent.